src/stepfunctions/inputs/placeholders.py

```diff
@@ -125,12 +125,7 @@ def contains(self, placeholder):
         Returns:
             bool: `True` if placeholder variable was found in the collection. `False`, otherwise.
         """
-        for k, v in self.store.items():
-            if placeholder == v:
-                return True
-            elif v.contains(placeholder):
-                return True
-        return False
+        return self.contains(placeholder)
```

Comment:
You should keep the old logic here, and add a new method called `__contains__` so that we can make `v not in step_output` work.

```python
    def __contains__(self, placeholder):
        return self.contains(placeholder)
```

Reply:
Apologies, I missed the dunder I've reinstated and added a new method